packages/client-adapters/src/adapters/lexicons/adapter.js

```diff
@@ -130,6 +131,68 @@ class AlpheiosLexiconsAdapter extends BaseAdapter {
       this.addError(this.l10n.messages.LEXICONS_NO_ALLOWED_URL)
       return
     }
+
+    if (this.async) {
+      return this.fetchDefsWithCallbacks(homonym, lookupFunction)
+    } else {
+      if (lookupFunction === 'short') {
+        return this.fetchShortDefsSync(homonym)
+      } else if (lookupFunction === 'full') {
+        await this.fetchFullDefsSync(homonym)
+      }
+    }
+  }
+
+  /**
+  * This is a sync method that retrieves short definitions for homonym synchronously
+  * @param {Homonym} homonym - homonym for retrieving definitions
+  */
+  async fetchShortDefsSync (homonym) {

```

Comment:
It is a little confusing, on my opinion, to have a sync function being declared as async. Maybe it's just me, but it took me a while to grasp it. If I understand correctly, this function retrieves multiple short definitions sequentially. So maybe it's worth to rename it, removing the `sync` part and adding some indication that it retrieves multiple definitions, in contrast with the existing function? What do you think?

Reply:
It could return several definitions or could return only one definition, it depends given options.

As for me, async is a special word that shows that this method is working with Promises, and it doesn't tell if actions would be done in parallel or in sequence, it could be understood from the following code.

I have used to type of function names:
 
`fetchDefsWithCallbacks` (here we have both calls for short and full)
`fetchShortDefsSync`
`fetchFullDefsSync`

`updateFullDefsWithCallbacks`
`updateFullDefsSync`

That's why I used this types of addition - `withCallbacks` and `sync`

but of course I am not very good with english, may be there are some other words that show
that the method executes all requests and wait untill all of them would be retrieved

@kirlat and @balmas , do you have better suggestion?

Reply:
@balmas , we already have 

```
fetchShortDefs
fetchFullDefs
```

they are used as methods from client-adapters and navigate request through code

so I couldn't rename methods this way

Reply:
@balmas and @kirlat , may be we will have then

**from client-adapters**
`fetchShortDefs`
`fetchFullDefs`

`fetchDefsAsync` - with callbacks
`fetchShortDefsSync` - sync way
`fetchFullDefsSync` - sync way

?